app.rb

```diff
@@ -0,0 +1,164 @@
+      puts "\nList of all the persons:"
+      @person.each_with_index do |person, index|
+        if person.is_a?(Student)
+          puts "#{index + 1}. [Student] Name: #{person.name}, age: #{person.age}, with ID: #{person.id}"
+        else
+          puts "#{index + 1}. [Teacher] Name: #{person.name}, age: #{person.age}, with ID: #{person.id}
+          and specialization: #{person.specialization}"
+        end
+      end
+    end
+  end
+
+  def create_a_person
+    print 'Do you want to create a student [1] or a teacher [2]? [Input number]: '
+    person = gets.chomp.to_i
+    case person
+    when 1
+      @person.push(create_student)
+      puts "\nThe student created successfully."
+    when 2
+      @person.push(create_teacher)
+      puts "\nThe teacher created successfully."
+    else
+      puts 'Please, select Student [1] or Teacher [2] only!'
+      create_a_person
+    end
+  end
+
+  def create_student
+    puts "\nCreate a new student"
+    print 'Enter age: '
+    student_age = gets.chomp.to_i
+    print 'Enter name: '
+    student_name = gets.chomp.to_s
+    print 'Has parrent permition? Enter [Y/N]: '
+    parent_permission = gets.chomp
+    case parent_permission
+    when 'y'
+      Student.new(student_name, student_age)
+    when 'n'
+      Student.new(student_name, student_age, parent_permission: false)
+    else
+      puts 'Please, select "y" or "n"'
+      create_student
+    end
+  end
+
+  def create_teacher
+    puts "\nCreate a new teacher"
+    print 'Enter age: '
+    teacher_age = gets.chomp.to_i
+    print 'Enter name: '
+    teacher_name = gets.chomp
+    print 'Enter specialization: '
+    specialization = gets.chomp
+    Teacher.new(teacher_name, teacher_age, specialization)
+  end
+
+  def create_a_book
+    puts "\nCreate a new book"
+    print "\nEnter book's title: "
+    title = gets.chomp
+    print "Enter book's author: "
+    author = gets.chomp
+    @book.push(Book.new(title, author))
+    puts "Book #{title} created successfully.\n"
+  end
+
+  def create_a_rental
+    puts 'Select a book from the following list by number'
+    @book.each_with_index { |book, index| puts "#{index + 1}) Title: #{book.title}, Author: #{book.author}" }
+
+    book_id = gets.chomp.to_i
+
+    puts 'Select a person from the following list by number (not id)'
+    @person.each_with_index do |person, index|
+      puts "#{index + 1}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+
+    person_id = gets.chomp.to_i
+
+    print 'Date: '
+    date = gets.chomp.to_s
+
+    rental = Rental.new(date, @book[book_id], @person[person_id])
+    @rentals << rental
+
+    puts 'Rental was created successfully'
+    sleep 0.75
+  end
+
+  def list_all_rentals
+    print 'ID of person: '
+    id = gets.chomp.to_i
+
+    puts 'Rentals:'
+    @rentals.each do |rental|
+      puts "Date: #{rental.date}, Book '#{book.title}' by #{book.author}" if rental.person == id
+    end
+    sleep 0.75
```

Comment:
- [ ] Let's work on this function when I create a Rental, I get a successful message that Rental was created, but when I want to see the list of created rentals I see nothing. See the image below

![rental](https://user-images.githubusercontent.com/53124548/158011740-92c2196d-fb53-43f0-864c-3d7481aa2697.png)

Reply:
- Hello @elisha2kyakpo1 I have finally fixed the bug

  - ![listAllRentals](https://user-images.githubusercontent.com/83566118/158036412-cba72862-e5dd-4aa8-8db8-64fe89044226.png)